src/js/components/pipelines/FilterField.jsx

```diff
@@ -0,0 +1,47 @@
+'use strict';
+var React = require('react'),
+    connect = require('react-redux').connect,
+    Immutable = require('immutable'),
+    assets = require('../../util/assets'),
+    getInVis = require('../../util/immutable-utils').getInVis,
+    Icon   = require('../Icon'),
+    showExpressionTextbox = require('../../actions/datasetActions').showExpressionTextbox;
+
+function mapStateToProps(state, ownProps) {
+  return {};
+}
+
+function mapDispatchToProps(dispatch, ownProps) {
+  return {
+    showExpressionTextbox: function(dsId, show, time) {
+      dispatch(showExpressionTextbox(dsId, show, time));
+    }
+  };
+}
+
+var FilterField = React.createClass({
+
+  propTypes: {
+    field: React.PropTypes.object.isRequired,
+    dsId:  React.PropTypes.number,
+  },
+
+  showTextbox: function(evt) {
+    var props = this.props,
+        show = true,
+        dsId = props.dsId,
+        time = 10;
+
+    this.props.showExpressionTextbox(dsId, show, time);
+
+  },
+
+  render: function() {
+    return (<Icon onClick={this.showTextbox} glyph={assets.filter} width="10" height="10" />);
```

Comment:
If you rebase to the latest `lyra2`, you'll see how I've introduced tooltips to these icons (e.g., with [`FieldType`](https://github.com/vega/lyra/commit/b7b364e1b396b63a4c9dd80d0f6af3c9011eac26#diff-767424686b9dcce6f942a1ca4d0823ae)). It would be good to add a tooltip to each of the new buttons you introduce as well.